actionview/lib/action_view/template/renderable.rb

```diff
@@ -4,22 +4,44 @@ module ActionView
   class Template
     # = Action View Renderable Template for objects that respond to #render_in
     class Renderable # :nodoc:
-      def initialize(renderable)
+      def initialize(renderable, &block)
         @renderable = renderable
+        @block = block
       end
 
       def identifier
         @renderable.class.name
       end
 
-      def render(context, *args)
-        @renderable.render_in(context)
-      rescue NoMethodError
+      def render(context, locals)
+        options =
+          if @renderable.method(:render_in).arity == 1
+            ActionView.deprecator.warn <<~WARN
+              Action View support for #render_in without options is deprecated.
+
+              Change #render_in to accept keyword arguments.
+            WARN
+
+            {}
+          else
+            { locals: locals }
+          end
+
+        @renderable.render_in(context, **options, &@block)
+      rescue NameError
         raise ArgumentError, "'#{@renderable.inspect}' is not a renderable object. It must implement #render_in."
       end
 
       def format
-        @renderable.try(:format)
+        if @renderable.respond_to?(:format)
+          @renderable.format
+        else
+          ActionView.deprecator.warn <<~WARN
+            #{@renderable.inspect} must respond to `#format` to be renderable
```

Comment:
Documentation in a bunch of places on this PR says `format` is optional, here we are telling it is required. Which is the right requirement? If we need format, why?

Reply:
I agree with you, the documentation should be specific about the fact that `#format` must be implemented. I'll make those changes now.

I've added this deprecation because the requirement for `#format` was never mentioned prior to https://github.com/rails/rails/pull/50622/files.

Is requiring that `#format` be implemented considered a breaking change, since there might be apps that have `#render_in` objects that don't define `#format`?

Reply:
On the other hand, the code _does not_ current require that `#format` be implemented thanks to https://github.com/rails/rails/pull/50622/files#r1445897182.

We could continue to treat `#format` as optional, in which case the deprecation would not be necessary, and the code could remain `@renderable.try(:format)`.

Does that seem preferred over a deprecation?

Reply:
Before #50622, `format` was already required. That PR made it optional, so there is no change in behavior there that needs to be deprecated.

Now, if #50622 changed from `@renderable.try(:format)` to `@renderable.format` we would need a deprecation. The opposite direction doesn't need a deprecation.

Reply:
@rafaelfranca thank you for clarifying. 

Since #50622 made that optional, I will change the documentation to reinforce that it is optional, then remove the deprecation.

Reply:
👍🏽